### DISCUSSION TITLE
drive() and settings() overlap

### DISCUSSION THREAD

**Discussion by MrGibbage** [pybricks/support#1365] [General]
I see that the settings() function for drivebases don't affect the speed in the drive() function. I haven't tested it, but I assume that the acceleration values in the settings() function **are** used by the drive() function? My question is why it is broken up this way. I feel like I am missing something. It seems to me either the drive function should take acceleration parameters, or no parameters and everything is set by the settings() function. Really, just curious, but mostly wondering if I am just missing something because it seems odd to have this partial overlap. In other words, what is the best way to use the drive function if we also want to control acceleration? Thanks in advance!

**Comment by laurensvalk** [comment]
Thanks for asking!

The `straight`, `turn` and `curve` functions use a default speed to make typical driving across a competition board quite convenient.

Setting the speed once in `settings` makes it easier to follow the rest of the program as the speed isn't repeated everywhere.

For `drive`, the speed and turn rate are typically adjusted all the time, like for line following. But more importantly, you have to specify a direction somehow, so that is done with the `drive_speed` and `turn_rate` arguments.

Indeed, acceleration affects all drive / turn / straight / curve functions.
